app/scripts/models/account.js

```diff
@@ -334,12 +339,35 @@ define(function (require, exports, module) {
         if (password) {
           return self._fxaClient.signIn(email, password, relier, {
             metricsContext: self._metrics.getActivityEventMetadata(),
-            reason: options.reason
+            reason: options.reason || SignInReasons.SIGN_IN,
+            resume: options.resume
           });
         } else if (sessionToken) {
           // We have a cached Sync session so just check that it hasn't expired.
           // The result includes the latest verified state
-          return self._fxaClient.recoveryEmailStatus(sessionToken);
+          return self._fxaClient.recoveryEmailStatus(sessionToken)
+            .then(function (sessionStatus) {
+              if (! sessionStatus.verified) {
+                // This is a little bit unnatural. /recovery_email/status
```

Comment:
Maybe push this down into the fxa-client so that the account doesn't have to worry about it.